native/src/main/java/tech/sourced/babelfish/JsonASTVisitor.java

```diff
@@ -578,6 +609,59 @@ public int visit(IASTName node) {
         return PROCESS_SKIP;
     }
 
+    @Override
+    public int visit(ICPPASTVirtSpecifier node) {
+        try {
+            json.writeStartObject();
+            try {
+                serializeCommonData(node);
+                String kindStr;
+                switch(node.getKind()) {
+                    case Final:
+                        kindStr = "final";
+                        break;
+                    case Override:
+                        kindStr = "override";
+                        break;
+                    default:
+                        kindStr = "unkown_virt_specifier_kind";
+                }
+                json.writeStringField("kind", kindStr);
+            } finally {
+                json.writeEndObject();
+            }
+        } catch (IOException e) {
+            enableErrorState(e);
+            return PROCESS_ABORT;
+        }
+        return PROCESS_SKIP;
+    }
+
+    @Override
+    public int visit(ICPPASTClassVirtSpecifier node) {
```

Comment:
Maybe not worth it—but can the duplicated code be parameterized? (I guess it might have to be generic and maybe not worthwhile for that)

Reply:
I was going to reply that, as a fan of DRY, I tough about it previously but having to return anyway from the wrapper, downcast the node type to IASTNode and upcast it inside the function was not really worth it in lines of code and definitely not in complexity of the code.

But after reading this comment I remembered Java fans celebrated that Java 8 finally got lambdas (_has science gone too far?_) so I decided to try them and arrived to a nice solution that will be on the next commit (PTAL) with this form:

```java
    @Override
    public int visit(IASTSomeType node) {
        return visitWrapper((IASTNode)node, () -> {
                // code without boilderplate
            }
            json.writeStringField("kind", kindStr);
        });
    }
```

So thanks!

Reply:
You're right, I hadn't thought through the need to cast in both directions—I'd been thinking in terms of a generic instantiated directly for the type.  It's up to you whether you think the savings is worth the indirection.

Reply:
The lambda solves it nicely, I think, since it gets the `node` variable of the enclosing context with the right type (`IASTSomeType` in the example) while the wrapper gets a node casted to `IASTNode` (because it doesn't need more for the generic parts).